### PR TITLE
did some final clean-up of the whole project

### PR DESCRIPTION
**Pull request by KseniiaMakarova**

### CODE REVIEW

File: src/main/java/com/internet/shop/controller/users/DeleteUserController.java [hunk 1]
```diff
@@ -25,15 +29,23 @@
...
-            throws IOException {
+            throws IOException, ServletException {
         Long userId = Long.valueOf(req.getParameter("id"));
-        ShoppingCart shoppingCart = shoppingCartService.getByUserId(userId);
-        shoppingCartService.delete(shoppingCart.getId());
-        List<Order> orders = orderService.getUserOrders(userService.get(userId));
-        for (Order order : orders) {
-            orderService.delete(order.getId());
+        if (userId.equals(req.getSession().getAttribute(USER_ID))) {
+            req.getRequestDispatcher("/views/accessDenied.jsp").forward(req, resp);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controller/users/DeleteUserController.java:35]: вже навіть самовипилитися не можна

File: src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 2]
```diff
@@ -107,54 +126,52 @@ public boolean delete(Long id) {
...
-            for (Product product : order.getProducts()) {
-                PreparedStatement insertStatement =
-                        connection.prepareStatement(insertOrdersProductsQuery);
-                insertStatement.setLong(1, order.getId());
-                insertStatement.setLong(2, product.getId());
-                insertStatement.executeUpdate();
-            }
+        for (Product product : order.getProducts()) {
+            PreparedStatement insertStatement =
+                    connection.prepareStatement(insertOrdersProductsQuery);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java:134]: let's move it before loop

File: src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java [hunk 3]
```diff
@@ -109,17 +113,31 @@ public boolean delete(Long id) {
...
     }
 
-    private void insertCartsProducts(ShoppingCart shoppingCart) throws SQLException {
+    private void insertCartsProducts(ShoppingCart shoppingCart, Connection connection)
+            throws SQLException {
         String insertCartsProductsQuery = "INSERT INTO carts_products (cart_id, product_id) "
                 + "VALUES (?, ?);";
+        for (Product product : shoppingCart.getProducts()) {
+            PreparedStatement insertStatement =
+                    connection.prepareStatement(insertCartsProductsQuery);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java:122]: same

File: src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java [hunk 4]
```diff
@@ -135,22 +120,34 @@ public boolean delete(Long id) {
         }
     }
 
-    private void insertUsersRoles(User user) throws SQLException {
+    private void insertUsersRoles(User user, Connection connection) throws SQLException {
         String selectRoleIdQuery = "SELECT role_id FROM roles WHERE role_name = ?";
         String insertUsersRolesQuery = "INSERT INTO users_roles (user_id, role_id) VALUES (?, ?);";
+        for (Role role : user.getRoles()) {
+            PreparedStatement selectStatement =
+                    connection.prepareStatement(selectRoleIdQuery);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java:128]: and here

File: src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java]: done)

File: src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java (hunk 3 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java]: done, thanks!

File: src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java (hunk 4 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: done)

File: src/main/java/com/internet/shop/controller/users/DeleteUserController.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/controller/users/DeleteUserController.java:35]: suicide is never a solution)

**Pull request merged**